src/overlays/actors/ovl_Obj_Wind/z_obj_wind.c

```diff
@@ -28,21 +29,144 @@ ActorInit Obj_Wind_InitVars = {
+    { 0x026C, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0654, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 }, { 0x0190, 0x0064, 0x07D0, 0x03E8, 0x0000 },
+    { 0x0258, 0x0078, 0x0BB8, 0x07D0, 0x0000 }, { 0x01F4, 0x0050, 0x0BB8, 0x07D0, 0x0000 },
+    { 0x0190, 0x003C, 0x07D0, 0x07D0, 0x0000 }, { 0x01D6, 0x0046, 0x07D0, 0x03E8, 0x0000 },
+};
+
+static s32 D_80B245CC = 0xFFFFFFFF;
+
+void ObjWind_Update(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    sD_80B2448C* entry = &D_80B2448C[(this->actor.params >> 7) & 0x1F];
+    Player* player;
+    f32 cosYTimesX; // sp80
+    f32 cosX;       // sp7C
+    f32 sinYTimesX; // sp78
+    Vec3f posCopy;  // sp6C;
+    Vec3f sp60;
+    Vec3f sp54;
+    f32 dist;     // sp50
+    f32 temp_fv0; // sp4C
+
+    if (D_80B245CC != DREG(85)) {
+        if ((DREG(85) >= 0) && (DREG(85) < 0x20)) {
+            DREG(80) = D_80B2448C[DREG(85)].unk_0;
+            DREG(81) = D_80B2448C[DREG(85)].unk_2;
+            DREG(82) = D_80B2448C[DREG(85)].unk_4;
+            DREG(83) = D_80B2448C[DREG(85)].unk_6;
+            DREG(84) = D_80B2448C[DREG(85)].unk_8;
+            D_80B245CC = DREG(85);
+        }
+    } else {
+        D_80B2448C[DREG(85)].unk_0 = DREG(80);
+        D_80B2448C[DREG(85)].unk_2 = DREG(81);
+        D_80B2448C[DREG(85)].unk_4 = DREG(82);
+        D_80B2448C[DREG(85)].unk_6 = DREG(83);
+        D_80B2448C[DREG(85)].unk_8 = DREG(84);
+    }
+    if (((this->actor.params & 0x7F) == 0x7F) || !Flags_GetSwitch(play, this->actor.params & 0x7F)) {
+        player = GET_PLAYER(play);
+        Math_Vec3f_Copy(&posCopy, &this->actor.world.pos);
+        cosX = Math_CosS(this->actor.shape.rot.x);
+        dist = Math_SinS(this->actor.shape.rot.x);
+        sinYTimesX = Math_SinS(this->actor.shape.rot.y) * dist;
+        cosYTimesX = Math_CosS(this->actor.shape.rot.y) * dist;
+        temp_fv0 = func_80179A44(&posCopy, &player->actor.world, &sp60);
```

Comment:
For reference, `func_80179A44` is called `Math3D_LineClosestToPoint` in the math3D branch